games/Snowman_Escape.js

```diff
@@ -0,0 +1,384 @@
+55555555555555..`],
+  [background, bitmap`
+2222222222222222
+2222222222221222
+2222222222222122
+2211222222222222
+2212222212222222
+2222222222222222
+2222222222222222
+2222222222222212
+2222222222222222
+2222122222222222
+2222212221122222
+2222222222122222
+2222222222222222
+2122222222222212
+2222222122222122
+2222221122222222`],
+  [goal, bitmap`
+.....DDDDDD.....
+...DDDDDDDDDD...
+..DDDDDDDDDDDD..
+.DDDDDDDDDDDDDD.
+.DDDDDDDDDDDDDD.
+DDDDDDDDDDDDDDDD
+DDDDDDDDDDDDDDDD
+DDDDDDDDDDDDDDDD
+DDDDDDDDDDDDDDDD
+DDDDDDDDDDDDDDDD
+DDDDDDDDDDDDDDDD
+.DDDDDDDDDDDDDD.
+.DDDDDDDDDDDDDD.
+..DDDDDDDDDDDD..
+...DDDDDDDDDD...
+.....DDDDDD.....`],
+  [toggle, bitmap`
+.....555555.....
+...5577777755...
+..577777777775..
+.57777777777775.
+.57777777777775.
+5777777777777775
+5777777777777775
+5777777777777775
+5777777777777775
+5777777777777775
+5777777777777775
+.57777777777775.
+.57777777777775.
+..577777777775..
+...5577777755...
+.....555555.....`],
+)
+
+setSolids([player,rock,machine,wallOne,wallTwo,wallThree])
+
+let level = 0;
+
+
+const levels = [
+  map`
+3233121
+213...g
+p.r..23
+1231m12`,
+  map`
+231221231
+1p.g.r.t2
+321231322`,
+  map`
+3213112
+2.g..t3
+1.pr..1
+1332m.2
+2212..2
+3121211`,
+  map`
+333333311
+3......g1
+3.r.....1
+333.....1
+3pm...1m1
+333333311`,
+  map`
+23333333331
+2.........1
+2.pr......3
+2.....mmm.2
+22231.....3
+t...rg....1
+2231m222222`,
+  map`
+211211
+2p..g3
+1....2
+11r.t2
+3...31
+3.mm.3
+1....1
+223122`,
+  map`
+231312211133
+2..........1
+3..........2
+1.pr....m..1
+1......m.m.1
+231122..m..3
+t....rg....1
+23213m311322`,
+  map`
+..........
+..........
+...m..m...
+..........
+.12331122.
+.1......1.
+..3....1..
+...1123...
+..........
+2312113331
+1p......g2
+1211332211`,
+];
+
+setMap(levels[level])
+
+setBackground(background)
+
+setPushables({
+  [player]: [rock, machine],
+  [rock]: [rock, machine],
+  [machine]: [machine, rock],
+});
+
+
+onInput("w", () => {
+  getFirst(player).y -= 1;
+});
+
+onInput("s", () => {
+  getFirst(player).y += 1;
+});
+
+onInput("a", () => {
+  getFirst(player).x -= 1;
+});
+
+onInput("d", () => { 
+  getFirst(player).x += 1;
+});
+
+onInput("j", () => {
+  const currentLevel = levels[level]; // get the original map of the level
+
+  // make sure the level exists before we load it
+  if (currentLevel !== undefined) {
+    clearText("");
+    setMap(currentLevel);
+    addLasers()
+  }
+});
+
+function addLasers() {
+  let machines = getAll(machine);
+
+  for (let object = 0; object < machines.length; object++) {
+    for (let i = 0; i < machines[object].y; i++) {
+      const x = machines[object].x
+      const y = machines[object].y - (1 + i)
+
+      const tile = getTile(x, y)
+      const playerb = getFirst(player)
+      const goalb = getFirst(goal)
+
+      if (tile.length == 0) {
+        addSprite(x, y, laser)
+      }
+        
+      /*There is definitely a better way to do this, */
+      /*but all things I tried didn't work because JS scares me*/
```

Comment:
Haha but JS isnt a snake like others ;)

Reply:
Haha lol, at least the Python looks pretty cool. Still wouldn’t want to come across one in person, though.